pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,377 @@
+    )    
+
+    steps = Instrument.control(
+        "VN", "N%i",
+        """An integer property that represents the number of steps the motor controller will run upon the next 'G' (go) 
+           command that is sent to the controller. This property can be set.""",
+        validator=truncated_range,
+        values=[0, 8388607],
+    )
+    
+    encoder_autocorrect = Instrument.control(
+        "VEA", "EA%i",
+        """A boolean property to enable or disable the encoder auto correct function. This property can be set.""",
+        map_values=True, 
+        values={True: 1, False: 0},
+        validator=strict_discrete_set,
+        get_process=lambda ea: int(ea),
+    )
+
+    encoder_delay = Instrument.control(
+        "VED", "ED%i",
+        """An integer property that represents the wait time in ms. after move is finished before
+           the encoder is read for a potential encoder auto-correct action to take place. This property can be set.""",
+        validator=truncated_range,
+        values=[0, 65535], 
+    )
+
+    encoder_motor_ratio = Instrument.control(
+        "VEM", "EM%i",
+        """An integer property that represents the ratio of the number of encoder pulses per motor step. 
+           This property can be set.""",
+        validator=truncated_range,
+        values=[1, 255],
+    )
+
+    encoder_retries = Instrument.control(
+        "VER", "ER%i",
+        """An integer property that represents the number of times the motor controller will try
+           the encoder auto correct function before setting an error flag. This property can be set.""",
+        validator=truncated_range,
+        values=[0, 255],
+    )
+
+    encoder_window = Instrument.control(
+        "VEW", "EW%i",
+        """An integer property that represents the allowable error in encoder pulses from the desired
+           position before the encoder auto-correct function runs. This property can be set.""",
+        validator=truncated_range,
+        values=[0, 255],
+    )
+
+    busy = Instrument.measurement(
+            "VF",
+            """Query to see if the controller is currently moving a motor."""
+    )
+    
+    error_reg = Instrument.measurement(
+        "!",
+        """Reads the current value of the error codes register.""",
+        get_process=lambda err: DPSeriesErrors(int(err)),
+    )
+
+    def check_errors(self):
+        """ Method to read the error codes register and log when an error is detected.
+        
+        :return error_code: one byte with the error codes register contents
+        """
+        current_errors = self.error_reg
+        if current_errors != 0:
+            logging.error("DP-Series motor controller error detected: %s" % current_errors)
+        return current_errors
+    
+    def __init__(self, resourceName, address=0, encoder_enabled=False, **kwargs):
+        """
+        Initialize communication with the motor controller with id=idn. In addition to the keyword arguments that can be
```

Comment:
What's this "id=idn" that is mentioned here? Left-over from a previous iteration?

Reply:
Thanks for catching this. Yes, this is leftover from a previous iteration.